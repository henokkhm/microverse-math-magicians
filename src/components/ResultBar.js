import PropTypes from 'prop-types';

import styles from '../styles/ResultBar/ResultBar.module.css';

function ResultBar({ displayValue }) {
  return <div data-testid="calculation-result" className={styles.resultBar}>{displayValue}</div>;
}

ResultBar.propTypes = {
  displayValue: PropTypes.string,
};

ResultBar.defaultProps = {
  displayValue: '0',
};
export default ResultBar;
