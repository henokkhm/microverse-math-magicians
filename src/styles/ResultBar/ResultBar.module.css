.resultBar {
  font-size: 1.15rem;
  font-weight: 500;
  width: 100%;
  padding: 1rem;
  color: rgb(255, 251, 251);
  background: #858694;
  text-align: right;
  margin-bottom: -1px;
}
