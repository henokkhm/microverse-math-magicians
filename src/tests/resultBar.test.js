import renderer from 'react-test-renderer';
import ResultBar from '../components/ResultBar';

describe('The resultBar', () => {
  it("renders 0 if we don't pass a display value", () => {
    const tree = renderer.create(<ResultBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the value passed', () => {
    const tree = renderer.create(<ResultBar displayValue="5" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the value passed', () => {
    const tree = renderer.create(<ResultBar displayValue="9" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
