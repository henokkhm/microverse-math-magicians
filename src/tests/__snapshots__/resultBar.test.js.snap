// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The resultBar renders 0 if we don't pass a display value 1`] = `
<div
  className="resultBar"
  data-testid="calculation-result"
>
  0
</div>
`;

exports[`The resultBar renders the value passed 1`] = `
<div
  className="resultBar"
  data-testid="calculation-result"
>
  5
</div>
`;

exports[`The resultBar renders the value passed 2`] = `
<div
  className="resultBar"
  data-testid="calculation-result"
>
  9
</div>
`;
